services/leanpub/leanpub-book-summary.service.js

```diff
@@ -0,0 +1,88 @@
+'use strict'
+
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+
+const bookSummarySchema = Joi.object({
+  id: Joi.number().required(),
+  page_count_published: Joi.number().required(),
+  total_copies_sold: Joi.number().required(),
+}).required()
+
+const keywords = ['leanpub']
+
+module.exports = class LeanpubBookSummaryService extends BaseJsonService {
+  static render({ label, message }) {
+    return {
+      label,
+      message,
+    }
+  }
+
+  static get defaultBadgeData() {
+    return { color: 'blue', label: 'leanpub' }
+  }
+
+  static get category() {
+    return 'other'
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Leanpub Book Page Count',
+        pattern: ':book/:metric',
+        namedParams: {
+          book: 'juice-shop',
+          metric: 'pages',
+        },
+        staticExample: this.render({ label: 'pages', message: 226 }),
+        keywords,
+      },
+      {
+        title: 'Leanpub Book Total Copies Sold',
+        pattern: ':book/:metric',
+        namedParams: {
+          book: 'juice-shop',
+          metric: 'sold',
+        },
+        staticExample: this.render({ label: 'sold', message: 2691 }),
+        keywords,
+      },
+    ]
+  }
+
+  static get route() {
+    return {
+      base: 'leanpub/book/summary',
+      format: '([^/]+)/(pages|sold)',
```

Comment:
Usually our URL schema is like

`npm/version/:package`
`github/tag/:user/:repo` etc

Seems like it would be more consistent to have

`leanpub/pages/:book` or
`leanpub/book/pages/:book` (if we might have badges for leanpub stuff that isn't books)

Also, you could define this using a `pattern` here instead of a regex and then you don't need to define it in the examples. So this could be


```js
static get route() {
  return {
    base: 'leanpub/book',
    pattern: ':book(pages|sold)/:metric',
  }
}
```

and then you can dispense with the `pattern` key in the examples.

Reply:
disclaimer: I've never used Leanpub myself :) In looking at their API docs it does seem like there would be the potential (however remote) for other Leanpub badges in the future not related to book metrics. 

I think I understand the suggestion and will update accordingly

Reply:
So I made the changes and it is functional, including removing the  `pattern` key from the examples:
```js
{
  title: 'Leanpub Book Page Count',
  namedParams: {
    metric: 'pages',
    book: 'juice-shop',
  },
  staticExample: this.render({ label: 'pages', message: 226 }),
  keywords,
},
```
But this results in an odd looking url showing on the examples page: 

`http://[::]:8080/leanpub/book/:metric(pages%7Csold)/:book.svg`

Is there another key I should add to the examples so that it will pull in the right metric? I.e. I'd assume the urls should look like: 

`http://[::]:8080/leanpub/book/pages/:book.svg` and
`http://[::]:8080/leanpub/book/sold/:book.svg`

Reply:
aah - I hadn't considered that. We probably need to improve that in the front-end, but lets not shave that yak in this PR. We can leave it as a regex for now, so we can define the route as:

```js
    base: 'leanpub/book',
    format: '(pages|sold)/([^/]+)',
    capture: ['metric', 'book']
```

and set a pattern in the examples

Reply:
I already pushed the change to use `pattern` in the route, and I added a modified `pattern` key back to the examples so they'd display correctly:

```js
{
  title: 'Leanpub Book Page Count',
  namedParams: {
    metric: 'pages',
    book: 'juice-shop',
  },
  pattern: 'pages/:book',
  staticExample: this.render({ label: 'pages', message: 226 }),
  keywords,
},
```

Will that work or would you prefer I switch back to the `format/capture` approach in the route?

Reply:
aah yeah - that works too. Even better, in fact :)

Reply:
> I added a modified `pattern` key back to the examples so they'd display correctly:

For what it's worth, that's the approach I've mostly been using.

I'm not sure how best to handle this in the long run. It's possible we could get to a point that we show one pattern per service class in the listing, and present the examples on the popup. The biggest blocker right now is pillarjs/path-to-regexp#173, though there are a few other cases as you've pointed out, where the regex is too complicated.